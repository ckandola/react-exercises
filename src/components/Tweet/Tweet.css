.tweet-date {
  font-weight: 600;
}

.tweet-message {
  font-style: italic;
}

.tweet-name {
  font-weight: bold;
  color:black;
}

.tweet-user {
  font-size: 0.8em;
  color:dimgrey;
}

.tweet-box-outer {
  background-color: rgb(29, 161, 242);
  max-width: 50vw;
  margin: auto;
  max-height:10em;
}

.tweet-box-inner {
  background-color: white;
}
