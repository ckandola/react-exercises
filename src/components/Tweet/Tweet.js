import React from "react";
import "./Tweet.css";
import { IoChatbubbleOutline } from 'react-icons/io5';
import { AiOutlineRetweet, AiOutlineHeart } from 'react-icons/ai';
import { FiUpload } from 'react-icons/fi';
import PropTypes from 'prop-types';

const Tweet = ({ username, name, date, message }) => {
  return (
    <>
      <div className="tweet-box-outer">
        <span className="tweet-span">
          <div className="tweet-name">{`${name.toUpperCase()}`}</div>
          <div className="tweet-separator">{'·'}</div>
          <div className="tweet-user">
            <a href={`https://twitter.com/${username}`}>
              @{username}
            </a>
          </div>
          <div className="tweet-separator">{'·'}</div>
          <div className="tweet-date">{`${date}`}</div>
        </span>
        <div className="tweet-box-inner">
          <div className="tweet-message">{message}</div>
          <span className="tweet-btn-span">
            <button className="tweet-btn">
              <IoChatbubbleOutline />
            </button>
            <button className="tweet-btn">
              <AiOutlineRetweet />
            </button>
            <button className="tweet-btn">
              <AiOutlineHeart />
            </button>
            <button className="tweet-btn">
              <FiUpload />
            </button>
          </span>
        </div>
      </div>
    </>
  );
};

export default Tweet;

Tweet.propTypes = {
  username: PropTypes.string,
  name: PropTypes.string,
  date: PropTypes.string,
  message: PropTypes.string,
};
