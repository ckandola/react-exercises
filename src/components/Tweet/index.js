import React from "react";
import "./Tweet.css";

const Tweet = ({ username, name, date, message }) => {
  return (
    <>
      <div>
        <span className="user">{username}</span>
        <span className="name">({name})</span>:
        <span className="message">{message}</span>
      </div>
      <p className="date">{`--- ${date}`}</p>
    </>
  );
};

export default Tweet;
