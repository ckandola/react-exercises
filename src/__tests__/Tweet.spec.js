import React from 'react';
import TweetContainer from '../components/Tweet';
import Tweet from '../components/Tweet/Tweet';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({adapter: new Adapter()});

describe('Tweet tests', () => {
    describe('TweetContainer tests', () => {
        const wrapper = shallow(<TweetContainer />);
        
        it('matches the snapshot for TweetContainer', () => {
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        
        it(`doesn't create a new Tweet when message is empty`, () => {
            wrapper.find('form').simulate('submit', { preventDefault () {} });
            expect(wrapper.find('Tweet')).toHaveLength(3);
        });
        
        it(`doesn't allow more than 280 characters to be added`, () => {
            let str = '';
            for(let i = 1; i <= 281; i++) {
                str += '1';
                wrapper.find('textarea').simulate('change', {target: {value: str}});
            }
            expect(wrapper.find('textarea').getElement().props.value).toHaveLength(280);
        });
        
        it(`creates a new Tweet when a message is written and the button is clicked`, () => {
            const textArea = wrapper.find('textarea');
            textArea.simulate('change', {target: {value: 'hello world'}});
            wrapper.find('form').simulate('submit', { preventDefault () {} });
            expect(wrapper.find('Tweet')).toHaveLength(4);
        });

    });

    it('matches the snapshot for Tweet', () => {
        const wrapper = shallow(
            <Tweet 
                username="captain_melonhead"
                name="JONNY 2X4"
                date={new Date('Nov 8 2009').toString()}
                message={"I'm going to miss Captain Melonhead. Plank says he'll miss Splinter the Wonderwood."}
            />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
