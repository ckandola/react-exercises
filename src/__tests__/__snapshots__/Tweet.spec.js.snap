// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweet tests TweetContainer tests matches the snapshot for TweetContainer 1`] = `
<section
  className="tweet-section"
>
  <form
    className="tweet-form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="username"
    >
      Username
    </label>
    <input
      id="username"
      onChange={[Function]}
      type="text"
      value="new_user"
    />
    <label
      htmlFor="name"
    >
      Display Name
    </label>
    <input
      id="name"
      onChange={[Function]}
      type="text"
      value="Rick"
    />
    <div
      className="tweet-message-input-box"
    >
      <label
        htmlFor="message"
      >
        Message
      </label>
      <textarea
        cols="30"
        id="message"
        name="message"
        onChange={[Function]}
        placeholder="Enter up to 280 characters"
        rows="3"
        value=""
      />
      <input
        className="tweet-message-btn"
        type="submit"
        value="Tweet"
      />
    </div>
  </form>
  <Tweet
    date="6/05/1999"
    key="0"
    message="Gravy!"
    name="ed"
    username="chickensrock"
  />
  <Tweet
    date="7/02/1999"
    key="1"
    message="If elected, inflation will be a thing of the gas!"
    name="eddy"
    username="Mucho-Dinero"
  />
  <Tweet
    date="7/05/1999"
    key="2"
    message="Messy, messy, messy!"
    name="edd"
    username="my_kingdom_for_sunscreen"
  />
</section>
`;

exports[`Tweet tests matches the snapshot for Tweet 1`] = `
<Fragment>
  <div
    className="tweet-box-outer"
  >
    <span
      className="tweet-span"
    >
      <div
        className="tweet-name"
      >
        JONNY 2X4
      </div>
      <div
        className="tweet-separator"
      >
        ·
      </div>
      <div
        className="tweet-user"
      >
        <a
          href="https://twitter.com/captain_melonhead"
        >
          @
          captain_melonhead
        </a>
      </div>
      <div
        className="tweet-separator"
      >
        ·
      </div>
      <div
        className="tweet-date"
      >
        Sun Nov 08 2009 00:00:00 GMT-0800 (Pacific Standard Time)
      </div>
    </span>
    <div
      className="tweet-box-inner"
    >
      <div
        className="tweet-message"
      >
        I'm going to miss Captain Melonhead. Plank says he'll miss Splinter the Wonderwood.
      </div>
      <span
        className="tweet-btn-span"
      >
        <button
          className="tweet-btn"
        >
          <IoChatbubbleOutline />
        </button>
        <button
          className="tweet-btn"
        >
          <AiOutlineRetweet />
        </button>
        <button
          className="tweet-btn"
        >
          <AiOutlineHeart />
        </button>
        <button
          className="tweet-btn"
        >
          <FiUpload />
        </button>
      </span>
    </div>
  </div>
</Fragment>
`;
